src/centos/scripts/start-marklogic.sh

```diff
@@ -322,10 +334,21 @@ else
 fi
 
 ################################################################
-# mark the node ready
+# check if node is available and mark it ready
 ################################################################
-info "Cluster config complete, marking this node as ready."
-sudo touch /var/opt/MarkLogic/ready
+HOST_RESP_CODE=""
+while true 
+do
+    HOST_RESP_CODE=$(curl http://"${HOSTNAME}":8001/admin/v1/timestamp -X GET -o /dev/null -s -w "%{http_code}\n" --anyauth --user "${ML_ADMIN_USERNAME}":"${ML_ADMIN_PASSWORD}")
+    if [ "${HOST_RESP_CODE}" -eq 200 ]; then
+        sudo touch /var/opt/MarkLogic/ready
+        info "Cluster config complete, marking this node as ready."
+        break
+    else
+        info "MarkLogic not ready yet, retrying."
```

Comment:
Can we include the HOST_RESP_CODE here with the information like:
"Status Code: ${HOST_RESP_CODE}. MarkLogic is not ready yet. Retry in 5 seconds."

Reply:
IMO including just the status_code might confuse/need to know which API is requested for the check. Also, we are not exposing that an API is used for checking whether MarkLogic is ready. Considering we have more than one way to verify readiness.